internal/webhook/util/serviceaccount.go

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import "strings"
+
+const (
+	// ServiceAccountUsernamePrefix is the service account username prefix
+	ServiceAccountUsernamePrefix = "system:serviceaccount:"
+	// ServiceAccountUsernameSeparator is the separator used in service account username.
+	ServiceAccountUsernameSeparator = ":"
+)
+
+// MatchesUsername checks whether the provided username matches the namespace and name
+// Use this when checking a service account namespace and name against a known string.
+func MatchesUsername(namespace, name, username string) bool {
```

Comment:
```suggestion
func ServiceAccountMatchesUsername(namespace, name, username string) bool {
```

To provide more meaning to the function name, when it's called as `utils.ServiceAccountMatchesUsername`. Also, please rename the package to `utils` instead of `util`, to remain consistent with other `utils` packages in the project.

Reply:
Done